jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -392,6 +394,17 @@ void checkAccess(Class<?> clazz) throws IllegalAccessException {
 		private void checkAccess(Class<?> targetClass, String name, int memberModifiers, MethodHandle handle) throws IllegalAccessException {
 			checkClassAccess(targetClass);
 
+			/*[IF Sidecar19-SE]*/
+			if (null != handle) {
+				MethodType type = handle.type();
+				Module accessModule = accessClass.getModule();
+				checkClassModuleVisibility(accessModule, targetClass, false);
+				checkClassModuleVisibility(accessModule, type.returnType, true);
```

Comment:
There are two calls to `checkClassModuleVisibility`, one setting `throwError` to false while the other says true.  Which is intended?

Reply:
The throwable depends on the context.  If the target class is not visible, we need to throw an IllegalAccessException. If the parameter or return types are not visible, we need to throw a IllegalAccessError.